packages/style-engine/class-wp-style-engine.php

```diff
@@ -293,6 +287,159 @@ protected static function is_valid_style_value( $style_value ) {
 		return true;
 	}
 
+	/**
+	 * Private constructor to prevent instantiation.
+	 */
+	private function __construct() {
+		foreach ( static::$stores as $store_key => $store_instance ) {
+			static::$stores[ $store_key ] = WP_Style_Engine_CSS_Rules_Store::get_store( $store_key );
+		}
+		// Register the hook callback to render stored styles to the page.
+		static::render_styles( array( __CLASS__, 'process_and_enqueue_stored_styles' ) );
+	}
+
+	/**
+	 * Utility method to retrieve the main instance of the class.
+	 *
+	 * The instance will be created if it does not exist yet.
+	 *
+	 * @return WP_Style_Engine The main instance.
+	 */
+	public static function get_instance() {
+		if ( null === static::$instance ) {
+			static::$instance = new static();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Stores a CSS rule using the provide CSS selector and CSS declarations.
+	 *
+	 * @param string $css_selector     When a selector is passed, the function will return a full CSS rule `$selector { ...rules }`, otherwise a concatenated string of properties and values.
+	 * @param array  $css_declarations An array of parsed CSS property => CSS value pairs.
+	 * @param string $store_key        A valid key corresponding to an existing store in static::$stores.
+	 *
+	 * @return string A compiled CSS string.
+	 */
+	public function store_css_rule( $css_selector, $css_declarations, $store_key ) {
```

Comment:
This could be the public method that layouts might use, or any other area that wants to store non-block-support-type styles.